### PR TITLE
Dev Cart development initial merge

### PR DESCRIPTION
**Pull request by hntejas**
Initial merge to main

Features included: 
1) Product listing page with filtering
2) Cart 
3) Wishlist
4) Landing Page

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      content="width=device-width, initial-scale=1, shrink-to-fit=no"
+    />
+    <meta name="theme-color" content="#000000" />
+    <!--
+      manifest.json provides metadata used when your web app is added to the
+      homescreen on Android. See https://developers.google.com/web/fundamentals/engage-and-retain/web-app-manifest/
+    -->
+    <link rel="manifest" href="%PUBLIC_URL%/manifest.json" />
+    <link rel="shortcut icon" href="%PUBLIC_URL%/favicon.ico" />
+    <!--
```
Comment (rohan9896) [inline public/index.html]: Remove these comments

File: public/index.html [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    <noscript>
+      You need to enable JavaScript to run this app.
+    </noscript>
+    <div id="root"></div>
+    <!--
+      This HTML file is a template.
+      If you open it directly in the browser, you will see an empty page.
+
+      You can add webfonts, meta tags, or analytics to this file.
+      The build step will place the bundled scripts into the <body> tag.
```
Comment (rohan9896) [inline public/index.html]: Remove unnecessary comments

File: src/components/Home/Home.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
+import { brands, categories } from "../../data";
+import { Link } from "react-router-dom";
+import "./home.css";
+
+export default function Home() {
+  return (
+    <>
+      <div className="home-banner-container">
+        {/* <img src={Banner} className="home-banner-img" /> */}
```
Comment (rohan9896) [inline src/components/Home/Home.js]: remove this in case you don't need it

File: src/styles.css [hunk 4]
```diff
@@ -0,0 +1,14 @@
+@import url("https://fonts.googleapis.com/css2?family=Roboto&display=swap");
+
+.App {
+  font-family: "Roboto", sans-serif;
+  text-align: center;
+}
+body {
+  /* background: #f1f3f6; */
```
Comment (rohan9896) [inline src/styles.css]: Remove this comment

File: src/components/Wishlist/Wishlist.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+import ProductList from "../ProductListing/ProductList/ProductList";
+import { WishlistContext } from "../../store/wishlist/wishlistContext";
+
+export default function Wishlist() {
+  const { wishlist } = useContext(WishlistContext);
+  return (
+    <>
+      <div className="wishlist-container">
+        <h2
+          style={{
```
Comment (rohan9896) [inline src/components/Wishlist/Wishlist.js]: remove this in-line css and write this in wishlist.css file

File: src/components/Home/Home.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        <div className="home-banner-content">
+          <h2>Hello Devs 👋</h2>
```
Comment (rohan9896) [inline src/components/Home/Home.js]: Emoji's should be wrapped in a span tag with role img like this- <span role="img" aria-label="hello">👋</span>

File: src/components/Wishlist/wishlist.css [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+@media (max-width: 761px) {
+  .wishlist-container {
+    flex-direction: column;
+  }
+  .product-list {
+    justify-content: space-evenly !important;
+  }
+
```
Comment (rohan9896) [inline src/components/Wishlist/wishlist.css:38]: Remove this space

**Review by rohan9896 — APPROVED** [review]
Hey tejas, I have reviewed your code and it was very clean and consistent, Here are some small changes you can make to make it even better.

**Comment by hntejas** [comment]
> Hey tejas, I have reviewed your code and it was very clean and consistent, Here are some small changes you can make to make it even better.

@rohan9896  Thanks for the review, I made all the suggested changes.

File: src/components/Auth/Login/Login.js [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+import { UserContext } from "../../../store/user/userContext";
+import * as userActionTypes from "../../../store/types/userActionType";
+import { showToast } from "../../../utils/helper";
+import "../auth.css";
+
+export default function Login({ from }) {
+  const validators = {
+    email: (email) => {
+      const re = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+      return re.test(String(email).toLowerCase());
```
Comment (kajol-rep) [inline src/components/Auth/Login/Login.js]: Instead of re you can give this variable a meaningful name.

**Review by kajol-rep — CHANGES_REQUESTED** [review]
Great work Tejas. Your app looks nice.
I have suggested a very small change you can do just for readability.

File: src/components/Auth/Login/Login.js (hunk 8 above)
Comment (hntejas) [reply src/components/Auth/Login/Login.js]: Hi Kajol, 

Thanks for taking time to review, I have made the necessary changes. Please check it once.

**Review by kajol-rep — APPROVED** [review]
Looks better now. You are good to go ahead.

**Pull request merged**